.github/workflows/tests.yml

```diff
@@ -20,17 +20,40 @@ jobs:
     steps:
     - name: Checkout
       uses: actions/checkout@v2
+      
     - name: Setup PHP
       uses: shivammathur/setup-php@v2
       with:
         php-version: ${{ matrix.php-versions }}
         extensions: gd, sqlite3
+      
+    # composer cache
     - name: Remove composer lock
       run: rm composer.lock
+    - name: Get composer cache directory
+      id: composer-cache
+      run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+    # http://man7.org/linux/man-pages/man1/date.1.html
+    # https://github.com/actions/cache#creating-a-cache-key
+    - name: Get Date
+      id: get-date
+      run: |
+        echo "::set-output name=date::$(/bin/date -u "+%Y%m%d")"
+      shell: bash
+    - name: Cache dependencies
+      uses: actions/cache@v2
+      with:
+        path: ${{ steps.composer-cache.outputs.dir }}
+        key: ${{ runner.os }}-composer-${{ steps.get-date.outputs.date }}-${{ hashFiles('**/composer.json') }}
```

Comment:
Okay, now using the json file and that works better of course. Now we could drop the daily cache purger again? Shall we?

Or is that bad?
AFAIK composer anyway seems to check on the run later, if the dependency is up-to-date and installs an update if required.

Reply:
In my experience with these free CI/CDs (travis, github) the caches aren't persisted for that long and get liberally purged when they need the space. They help when you re-run your action several times an hour, but after a day or two they usually have already been aged out.

Also, in a good caching scheme, you should not have to purge it until you change something in the code (a new dependency added or an update of a dependency). See for example our unit test image, where the container publication represents a cache and [every push triggers a cache refresh](https://github.com/PrivateBin/docker-unit-testing/commits/master) by rebuilding the image.

Reply:
> GitHub will remove any cache entries that have not been accessed in over 7 days. There is no limit on the number of caches you can store, but the total size of all caches in a repository is limited to 5 GB. If you exceed this limit, GitHub will save your cache but will begin evicting caches until the total size is less than 5 GB.

https://docs.github.com/en/actions/advanced-guides/caching-dependencies-to-speed-up-workflows#usage-limits-and-eviction-policy

Soo their policy is quite liberal, they seem to have plenty of space… :sweat_smile: